l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -819,9 +831,18 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         {
             // Inner call to encode data to decrease local var numbers
             _assetId = _ensureTokenRegisteredWithNTV(_l1Token);
+            IERC20(_l1Token).safeIncreaseAllowance(address(nativeTokenVault), _amount);
+        }
 
-            // solhint-disable-next-line func-named-parameters
-            bridgeMintCalldata = abi.encode(_amount, _prevMsgSender, _l2Receiver, getERC20Getters(_l1Token), _l1Token);
+        {
```

Comment:
This solution works, and it is quite intuitive. If we could increase the ChainBalance on the NTV from the L1ERC20 or AssetRouter directly it would be cheaper. But this is good.

Reply:
We could allow the bridge to increase chain balance via a separate method but it would look less straightforward imo